[Ng Wen Hao Dennis] iP

Denniszedead opened the pull request:
# BIG TECH
> How may I help? 😄 
## About
Big Tech is a revolutionary task list for a corporate leader. It:
- Creates the following tasks, TODO, Deadlines and Event.
- Shows the tasks in the list.
- Deletes the tasks in the list.
- Marks and unmarks the tasks.
## Why use?
- IT'S **FREE** (Duh, do i have to emphasise it further!)
- IT'S ~~Fast~~ **SUPER FAST**
## How to install
1. First, download the .jar [File]()
2. The from the directory where you store the .jar file, enter the following command in the terminal:
     `java -jar ip.jar`
3. Then, BIG TECH will start.
## Have you done all of them ?
- [ ] Download the .jar file?
## How to use
BIG TECH currently supports the following command:
- `` list ``: Shows all the tasks in the task list.
- `` todo <Name of the task>``: Creates a todo task.
## In case you are wondering
The main file is located here:
```java
public static void main(String[] args) {
        // Creates a new duke application and tells it to run.
        new Duke("data/duke.txt").run();
}
```

Commit:
Add Gradle support

Denniszedead committed:
Added binary files and temporary files into .gitignore

Denniszedead committed:
Completed Greet, Echo and Exit feature

Denniszedead committed:
Added add and list functions

Denniszedead committed:
Added Mark as Done feature

Denniszedead committed:
Added Todo, Events and Deadlines features

Denniszedead committed:
Changed method from "printEndRun" to "printAddEndRun"

Denniszedead committed:
Added Automated Text UI Testing

Denniszedead committed:
Added error handling functionality

Denniszedead committed:
Added delete functionality

Denniszedead committed:
Modified code with java enum

Denniszedead committed:
Change ItemList to TaskList

Denniszedead committed:
Remove all auto TODOs comments

Denniszedead committed:
Change ItemList.java to TaskList.java

Denniszedead committed:
Revert "Remove all auto TODOs comments"

This reverts commit ee1323dcb335ba91991520d07d4dfdb540a9333b.

Denniszedead committed:
Remane ItemList.java to TaskList.java

Denniszedead committed:
Merge branch 'master' into branch-Level-7

Denniszedead committed:
Implement Save feature

Denniszedead committed:
Add date time functionality for deadline and event

Denniszedead committed:
Remove todo comments

Denniszedead committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Task.java
#	src/main/java/TaskList.java

Denniszedead committed:
Fix date format on event tasks

Denniszedead committed:
Implement More OOP

Denniszedead committed:
Divide all classes into package

Denniszedead committed:
Move all source files to duke folder

Denniszedead committed:
Move all source code files to duke folder

Denniszedead committed:
Add JUnit testing

Denniszedead committed:
Package ip as jar file

Denniszedead committed:
Add JavaDocs

Denniszedead committed:
Ammend codes to follow Java Code Standard

Denniszedead committed:
Change all javadoc of run in all subclasses of comand with @inheritDocs

Denniszedead committed:
Implement find command

Denniszedead committed:
Remove data from merge

Denniszedead committed:
Revert "Change all javadoc of run in all subclasses of comand with @inheritDocs"

This reverts commit 7a669abae2b2341d88e215869c5a2971ada0a5ef.

Denniszedead committed:
Revert "Package ip as jar file"

This reverts commit ab6af426f0b3a810d0fc7c86f284521ed0e337ce.

Denniszedead committed:
Add data/duke.txt

Denniszedead committed:
Solve merge conflicts

Denniszedead committed:
Solve merge conflicts

Denniszedead committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/ui/Ui.java

Denniszedead committed:
Fix import standards on Command.java

Denniszedead committed:
Fix duke.txt

Denniszedead committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/dukeexceptions/DukeExceptions.java
#	src/main/java/duke/dukeexceptions/InvalidCommand.java
#	src/main/java/duke/praser/Parser.java
#	src/main/java/duke/storage/Storage.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/ui/Ui.java
#	src/main/java/duke/uicomponents/DeleteTaskDisplay.java
#	src/main/java/duke/uicomponents/ErrorMenu.java
#	src/main/java/duke/uicomponents/ExitScreen.java
#	src/main/java/duke/uicomponents/MarkTaskDisplay.java
#	src/main/java/duke/uicomponents/NewTaskDisplay.java
#	src/main/java/duke/uicomponents/UnmarkedTaskDisplay.java

Denniszedead committed:
Add comment header to parser class

hieunm1821 commented on src/main/java/duke/dukeexceptions/DukeExceptions.java:
```diff
@@ -0,0 +1,11 @@
+package duke.dukeexceptions;
```
I suggest the naming of the package is duke.exceptions

Denniszedead replied:
Duke exception is a specialised exception for duke so I actually prefer to name it that way.

hieunm1821 commented on src/main/java/duke/dukeexceptions/EmptyDate.java:
```diff
@@ -0,0 +1,21 @@
+/**
+ * The exception for when no date is entered for commands that requires dates.
+ */
+public final class EmptyDate extends DukeExceptions {
```
Should the name of this exception be EmptyDateException ? :thinking:

Denniszedead replied:
Yeap! You are right! I'll make the changes!

hieunm1821 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,80 @@
+        while (fileInput.hasNextLine()) {
+            // Gets the properties for a task.
+            String type = fileInput.nextLine();
+            Boolean done = Boolean.parseBoolean(fileInput.nextLine());
```
Should the name of this boolean variable be isDone?

Denniszedead replied:
Thanks! Must have overlooked it!

hieunm1821 reviewed (commented):
Overall, I found you code easy to read. There is no code quality issues. I noted some coding standard related issues only. Good work. :+1:

yumengtan commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,133 @@
+package duke.tasklist;
```
Like how u organize the classes into respective packages! 👍

yumengtan commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,80 @@
+/**
+ * The Storage object acts as an interface between Duke and the filename for database management.
+ */
+public class Storage{
```
Perhaps there should be a " " separator before the {. I have noticed that in ListIndexOutOfBound class as well!

Denniszedead replied:
Yeap! Must have overlooked it! Thanks for the notice!

yumengtan commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,80 @@
+        while (fileInput.hasNextLine()) {
+            // Gets the properties for a task.
+            String type = fileInput.nextLine();
+            Boolean done = Boolean.parseBoolean(fileInput.nextLine());
```
Perhaps using a prefix such as is to the variable name such as IsDone like @hieunm1821 mentioned would make it sound more like booleans!

yumengtan reviewed (commented):
Overall your code structure is very well organised! Just some nits to fix, well done! 👍

Denniszedead committed:
Address suggestions from reviewers

Denniszedead committed:
Merge branch 'barnch-A-Gradle'

* barnch-A-Gradle:
  Add Gradle support

Denniszedead committed:
Add gradle support

Denniszedead committed:
Add checkstyle

Denniszedead committed:
Revert "Add gradle support"

This reverts commit c37952b61e8fd893aff270fbacfdcc13fa902ab9.

Denniszedead committed:
Add CheckStyle

Denniszedead committed:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  Add CheckStyle

Denniszedead committed:
Implement FXML GUI

Denniszedead committed:
Add comments on DukeCommandTest

Denniszedead committed:
Change to v2.0

Denniszedead committed:
Add assertions

No assertions is present in the code

Assertions allow detection of bugs by the programmer to be detected
earlier.

Let's
* Add assertions to images that should be present in MainWindow.java
* Add assertions to files in Main.java, Duke.java and Storage.java

Denniszedead committed:
Change codes to follow coding quality

Some parts of the code did not follow the code quality according taught
in week 5. Following code quality will improve readebility for other
software developers.

Let's look through the code and change the code to following the
code quality learnt from Week 5.

Denniszedead committed:
Add help command

Duke does not have a help command.

Lets add help to the Command.java file to show all help for all commands
in duke.

Denniszedead committed:
All create task commands: Create detection of duplicated dates

There are currently no checks on the task list to ensure that is a date
for a task before a new task is created.

To simulate real life events, two unrelated events cannot happen at the
same time. Hence it is important that we have this mechanism to check
that an event with the same date has been created before creation.

Let's create the algorithm on Deadline and Event to check whether there
is a deadline/event with the same date before task creation.

Denniszedead committed:
Add Continuous Integration using Github Actions

Denniszedead committed:
Move gradle.yml to .github/workflows

Denniszedead committed:
Update gradle.yml

Denniszedead committed:
Merge pull request #1 from Denniszedead/branch-A-CI

Merge CI

Denniszedead committed:
Merge pull request #2 from Denniszedead/branch-A-Assertions

Add assertions

Denniszedead committed:
Merge branch 'master' into branch-A-CodeQuality

Denniszedead committed:
Merge pull request #3 from Denniszedead/branch-A-CodeQuality

Change codes to follow coding quality

Denniszedead committed:
Merge branch 'master' into branch-B-DetectAnomalies

Denniszedead committed:
Merge branch 'branch-B-DetectAnomalies' of https://github.com/Denniszedead/ip into branch-B-DetectAnomalies

* 'branch-B-DetectAnomalies' of https://github.com/Denniszedead/ip:
  Update gradle.yml
  Move gradle.yml to .github/workflows
  Add Continuous Integration using Github Actions
  Change codes to follow coding quality
  Add assertions

Denniszedead committed:
Rectify merge conflicts

Denniszedead committed:
Merge pull request #4 from Denniszedead/branch-B-DetectAnomalies

All create task commands: Create detection of duplicated dates

Denniszedead committed:
Merge branch 'master' of https://github.com/Denniszedead/ip

* 'master' of https://github.com/Denniszedead/ip:
  Rectify merge conflicts
  Update gradle.yml
  Move gradle.yml to .github/workflows
  Add Continuous Integration using Github Actions
  All create task commands: Create detection of duplicated dates
  Change codes to follow coding quality
  Add assertions

Denniszedead committed:
Resolve merge conflicts

Denniszedead committed:
Merge pull request #5 from Denniszedead/branch-C-Help

Branch c help

Denniszedead committed:
Merge branch 'master' of https://github.com/Denniszedead/ip

* 'master' of https://github.com/Denniszedead/ip:
  Add help command

Denniszedead committed:
Update version to v2.1

Denniszedead committed:
Improve GUI

The current GUI does not reflect user duke interaction. There is no
clear distinction between user and duke region. Furthermore, when
using the list command, the dialogue box did not show the full task
list.

Users will not be able to distinguish between user and duke interaction
easily. Furthermore, users will not be able to see all the tasks in the
task list.

Let's:
* Create an response dialogue box for responses from duke
* Create an user dialogue box for duke
* Set VBox in scroll pane, and all its children's height to
SET_COMPUTE
* Add different colour background for both response and user dialogue

Denniszedead committed:
Merge pull request #7 from Denniszedead/branch-A-BetterGui

Improve GUI

Denniszedead committed:
Fix adding deadline bug

Denniszedead committed:
Add User Guide

The current user guide is not defined properly.

New users will not understand how to use PCPrincipal.

Let's:
* Add instructions on how to start PCPrincipal.
* Add instructions for all commands in PCPrincipal.

Denniszedead committed:
Merge pull request #8 from Denniszedead/branch-A-UserGuide

Add User Guide

Denniszedead committed:
Merge branch 'master' of https://github.com/Denniszedead/ip

* 'master' of https://github.com/Denniszedead/ip:
  Add User Guide

Denniszedead committed:
Add Ui.png

Denniszedead committed:
Update README.md

Denniszedead committed:
Merge pull request #9 from Denniszedead/branch-A-UserGuide

Update README.md

Denniszedead committed:
Merge branch 'master' of https://github.com/Denniszedead/ip

* 'master' of https://github.com/Denniszedead/ip:
  Update README.md

Denniszedead committed:
Deploy to v3.0

Denniszedead committed:
Update link of PCPrincipal.jar

Denniszedead committed:
Merge pull request #10 from Denniszedead/branch-A-UserGuide

Update link of PCPrincipal.jar

Denniszedead committed:
Update README.md

Denniszedead committed:
Merge branch 'master' of https://github.com/Denniszedead/ip

* 'master' of https://github.com/Denniszedead/ip:
  Update README.md
  Update link of PCPrincipal.jar

Denniszedead committed:
Fix no file bug

Denniszedead committed:
Change build name to PCPrincipal

Denniszedead committed:
Update link in README.md

Denniszedead committed:
Update README.md

Denniszedead committed:
Update README.md

Denniszedead committed:
Fix jar build

Denniszedead committed:
Merge branch 'master' of https://github.com/Denniszedead/ip

* 'master' of https://github.com/Denniszedead/ip:
  Update README.md
  Update README.md
